src/components/shortcuts/use-shortcut.ts

```diff
@@ -0,0 +1,88 @@
+import { useCallback, useEffect, useRef } from 'react'
+import { useModalOpeners } from '../../contexts/ModalContext'
+import { useChatContext } from '../../contexts/ChatContext'
+import { useContentContext } from '@/contexts/ContentContext'
+import { shortcuts } from './shortcutDefinitions'
+
+function useAppShortcuts() {
+  const { setIsNewDirectoryModalOpen, setIsFlashcardModeOpen, setIsSettingsModalOpen } = useModalOpeners()
+  const { setSidebarShowing } = useChatContext()
+  const { createUntitledNote } = useContentContext()
+
+  const handleShortcut = useCallback(
+    (action: string) => {
+      switch (action) {
+        case 'open-new-note':
+          createUntitledNote()
+          break
+        case 'open-new-directory-modal':
+          setIsNewDirectoryModalOpen(true)
+          break
+        case 'open-search':
+          setSidebarShowing('search')
+          break
+        case 'open-files':
+          setSidebarShowing('files')
+          break
+        case 'open-chat-bot':
+          setSidebarShowing('chats')
+          break
+        case 'open-flashcard-quiz-modal':
+          setIsFlashcardModeOpen(true)
+          break
+        case 'open-settings-modal':
+          setIsSettingsModalOpen(true)
+          break
+        default:
+          // No other cases
+          break
+      }
+    },
+    [createUntitledNote, setIsNewDirectoryModalOpen, setSidebarShowing, setIsFlashcardModeOpen, setIsSettingsModalOpen],
+  )
+
+  const handleShortcutRef = useRef(handleShortcut)
+  handleShortcutRef.current = handleShortcut
+  // shortcuts.forEach(shortcut => {
+  //   useHotkeys(shortcut.key, (event) => {
+  //     event.preventDefault()
+  //     handleShortcut(shortcut.action)
+  //   }, {
+  //     enableOnFormTags: true,
+  //     preventDefault: true,
+  //     enabled: true // Ensure the hotkey is always enabled
+  //   }, [handleShortcut]) // Add handleShortcut to the dependency array
+  // })
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      const modifierPressed = event.ctrlKey || event.metaKey
+      const keyPressed = event.key.toLowerCase()
+
+      const triggeredShortcut = shortcuts.find((s) => {
+        const [mod, key] = s.key.toLowerCase().split('+')
+        return mod === 'mod' && modifierPressed && key === keyPressed
+      })
+
+      if (triggeredShortcut) {
+        event.preventDefault()
+        handleShortcutRef.current(triggeredShortcut.action)
+      }
+    }
+
+    window.addEventListener('keydown', handleKeyDown)
+    return () => {
+      window.removeEventListener('keydown', handleKeyDown)
+    }
+  }, [])
```

Comment:
it's a intended behavior because:
The handleKeyDown function is defined inside the useEffect, so it doesn't depend on any external variables that might change.
We want to set up global event listeners like this only once for the lifetime of the component.